### PR TITLE
Added JmeSurfaceView.java class , for embedding jme-game in a custom xml UI-design.

### PR DESCRIPTION
**Pull request by Scrappers-glitch**
Embedded jme Game on android view to manipulate jme games on custom xml UI designs , could be used within :

* android.app.Activity;
* androidx.appcompat.app.AppCompatActivity; or ComponentActivity.
* androidx.appcompat.app.AlertDialog;
* android.app.Fragment;
* androidx.appcompat.app.Fragment;

OR any OGLES Compatible Context based container.

### CODE REVIEW

**Comment by tlf30** [comment]
Can you please ensure formatting is correct. I see several formatting errors.

**Comment by Scrappers-glitch** [comment]
> Embedded jme Game on android view to manipulate jme games on custom xml UI designs , could be used within :
> *  android.app.Activity;
> * androidx.appcompat.app.AppCompatActivity; or ComponentActivity.
> * androidx.appcompat.app.AlertDialog;
> * android.app.Fragment;
> * androidx.appcompat.app.Fragment;

> OR any OGLES Compatible Context based container.

**Comment by Scrappers-glitch** [comment]
> Can you please ensure formatting is correct. I see several formatting errors.

Yep , sure , which formatting errors ? the title or in code ?

**Comment by Scrappers-glitch** [comment]
Okay, assuming the 'formatting errors' are the extra white spaces between the if statement conditions & methods , i have removed them , you can refer to a specific if you find something wrong.

**Comment by stephengold** [comment]
The JMonkeyEngine project plans to move toward the [Google Style Guide](https://google.github.io/styleguide/javaguide.html#s4-formatting) which requires whitespace after `if` and on both sides of binary operators such as "!=", "+", and "=". See the discussion at #1056.

File: jme3-android/src/main/java/com/jme3/app/JmeSurfaceView.java [hunk 1]
```diff
@@ -0,0 +1,467 @@
+package com.jme3.app;
```
Comment (stephengold) [inline jme3-android/src/main/java/com/jme3/app/JmeSurfaceView.java:1]: Please add the JME copyright notice and license.

**Comment by stephengold** [comment]
If @Scrappers-glitch concurs, I could run the new file through a formatter before integrating this PR. That might be easier (for everyone) than trying to format the code by hand.

**Comment by Scrappers-glitch** [comment]
> If @Scrappers-glitch concurs, I could run the new file through a formatter before integrating this PR. That might be easier (for everyone) than trying to format the code by hand.

I will reformat the code now ,& review it again , no worry.👍

**Comment by Scrappers-glitch** [comment]
I think it's better to change the annotations NonNullable from org.jetbrains to androidx.annotations package , so we are on the safe side , & reformat the code, then the PR would be ready to be merged.

**Comment by Scrappers-glitch** [comment]
So @stephengold , 
this : 
```java
        if ( legacyApplication != null ){
```
or this : 
```java 
        if (legacyApplication != null){
```

File: jme3-android/src/main/java/com/jme3/app/JmeSurfaceView.java (hunk 1 above)
Comment (Scrappers-glitch) [reply jme3-android/src/main/java/com/jme3/app/JmeSurfaceView.java:1]: > So @stephengold ,
> this :
> 
> ```java
>         if ( legacyApplication != null ) {
> ```
> 
> or this :
> 
> ```java
>         if (legacyApplication != null) {
> ```

**Comment by Ali-RS** [comment]
I think `if (legacyApplication != null) {` style is what mostly used in JME.

**Comment by Scrappers-glitch** [comment]
> I think `if (legacyApplication != null) {` style is what mostly used in JME.

Thank you i ensured that , one another question though about the interfaces , so having them in the same class or different files are better ? , because i like to have them at the same file since they are used only by this file.

**Comment by pspeed42** [comment]
I don't see any interfaces defined in this code.

I do see some scary "synchronized" keywords, though.  At least one seems unnecessary because nothing in the method requires synchronization.

The one on update() worries me, though.  Is there a chance that update() will be called from multiple threads at the same time?

Note: it's very very important that JME's update() is called from the same thread every time.  It could break some applications if it isn't.

**Comment by Scrappers-glitch** [comment]
@pspeed42 thanks for reviewing .

> I don't see any interfaces defined in this code.

At the end of the code , you will find them , anyway i have separated them into files , to be with jme style.



> The one on update() worries me, though. Is there a chance that update() will be called from multiple threads at the same time?

That was a long story of using FutureTasks to delay the add of the GlSurfaceView to the users ' screen without delaying jme-Render & it gets to be successful when using synchronized() because that basically makes jme SystemListener gets notified of the changes in the UI thread , but then i discovered a better way which is using a Handler & postDelay it , i see it's better optimized for android UI , the reason for all this , is that it gives an option for the user to use a SplashScreen while loading the scene asynchronously , so after your comment i tested it again w/o synchronized & it seems to be the same so i removed it as no use of it now & they make confusions as well.

if you donot want to delay the add of glsurfaceView use ```startRenderer(JmeSurfaceView.NO_DELAY);``` , but it's better to at least use dealy of 150 ms , again this doesn't delay the renderer , it's not even related to jme.

The legacyApplication update() method gets updated by the ```SystemListener``` interface , UI isn't involved in the initialize() or update() of jme game.

All the new updates with reformating code would be in the next commit.

**Comment by pspeed42** [comment]
Mmm... nested in with the methods like that, I missed the interfaces.

Inner classes/interfaces should go at the bottom.

In this case, it seems like you only use interfaces to avoid having the application extend this class?  It does feel weird to have them in the class like this... which is usually a sign that something is off.

**Comment by Scrappers-glitch** [comment]
> In this case, it seems like you only use interfaces to avoid having the application extend this class? It does feel weird to have them in the class like this... which is usually a sign that something is off.

i like to enclose all things that are used by each other inside one enclosure , so the answer because they are used by this class only , but anyway i have separated them into files as jme does.

**Comment by Scrappers-glitch** [comment]
> > In this case, it seems like you only use interfaces to avoid having the application extend this class? It does feel weird to have them in the class like this... which is usually a sign that something is off.
> 
> i like to enclose all things that are used by each other inside one enclosure , so the answer because they are used by this class only , but anyway i have separated them into files as jme does.



> Mmm... nested in with the methods like that, I missed the interfaces.

& As you quoted here , someone may literally miss them in the mess of methods.

**Comment by Ali-RS** [comment]
By the way, if you use IntelliJ IDE do a `Code -> Reformat Code`. That will fix the issue with white spaces.

**Comment by Scrappers-glitch** [comment]
> By the way, if you are on IntelliJ IDE do a `Code -> Reformat Code` that will fix the formatting issues.

I know , i have taken the time to fix them myself to be able to retain the way that i would use with jme style , the new commits are done.

**Review by Ali-RS — CHANGES_REQUESTED** [review]

**Comment by stephengold** [comment]
It's too late to make changes to this PR. If there's something you want to change, make a new PR.

**Pull request merged**